extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpIntegrationTests.java

```diff
@@ -0,0 +1,696 @@
+     *
+     * @param statusCode   Response status code.
+     * @param responseBody Response body.
+     * @return see {@link HttpResponse}
+     */
+    private HttpResponse withResponse(HttpStatusCode statusCode, String responseBody) {
+        var response = response()
+                .withStatusCode(statusCode.code());
+
+        if (responseBody != null) {
+            response.withHeader(
+                            new Header(HttpHeaderNames.CONTENT_TYPE.toString(),
+                                    MediaType.PLAIN_TEXT_UTF_8.toString())
+                    )
+                    .withBody(responseBody);
+        }
+
+        return response;
+    }
+
+    /**
+     * Mock HTTP POST request for sink.
+     *
+     * @param responseBody Request body.
+     * @return see {@link HttpRequest}
+     */
+    private HttpRequest givenPostRequest(String responseBody) {
+        return request()
+                .withMethod(HttpMethod.POST.name())
+                .withPath("/" + DPF_HTTP_API_PART_NAME)
+                .withHeaders(
+                        new Header(HttpHeaderNames.CONTENT_TYPE.toString(),
+                                MediaType.APPLICATION_OCTET_STREAM.toString()),
+                        new Header(AUTH_HEADER_KEY, SINK_AUTH_VALUE)
+                )
+                .withBody(binary(responseBody.getBytes(StandardCharsets.UTF_8)));
+    }
+
+    /**
+     * Mock response from sink.
+     *
+     * @param statusCode Response status code.
+     * @return see {@link HttpResponse}
+     */
+    private HttpResponse withResponse(HttpStatusCode statusCode) {
+        return withResponse(statusCode, null);
+    }
+
+    /**
+     * Mock error response which to force the connection to be dropped without any response being returned.
+     *
+     * @return see {@link HttpError}
+     */
+    private HttpError withDropConnection() {
+        return error()
+                .withDropConnection(true);
+    }
+
+    /**
+     * Provides most common http error status codes.
+     *
+     * @return Http Error codes as {@link Stream} of {@link Arguments}.
+     */
+    private static Stream<Arguments> provideCommonErrorCodes() {
+        return Stream.of(
+                Arguments.of(HttpStatusCode.MOVED_PERMANENTLY_301.name(), HttpStatusCode.MOVED_PERMANENTLY_301),
+                Arguments.of(HttpStatusCode.FOUND_302.name(), HttpStatusCode.FOUND_302),
+                Arguments.of(HttpStatusCode.BAD_REQUEST_400.name(), HttpStatusCode.BAD_REQUEST_400),
+                Arguments.of(HttpStatusCode.UNAUTHORIZED_401.name(), HttpStatusCode.UNAUTHORIZED_401),
+                Arguments.of(HttpStatusCode.NOT_FOUND_404.name(), HttpStatusCode.NOT_FOUND_404),
+                Arguments.of(HttpStatusCode.INTERNAL_SERVER_ERROR_500.name(), HttpStatusCode.INTERNAL_SERVER_ERROR_500),
+                Arguments.of(HttpStatusCode.BAD_GATEWAY_502.name(), HttpStatusCode.BAD_GATEWAY_502)
+        );
+    }
+
+}
```

Comment:
is a 700 line class really needed?

Reply:
Yeah, its too much. Already looking for idea to reduce it.
Suggestions are welcome.

Reply:
Did some refactoring as discussed.